packages/docregistry/src/context.ts

```diff
@@ -596,6 +596,11 @@ export class Context<
 
     try {
       await this._manager.ready;
+      if (!this.canSave) {
+        // File cannot be saved.
+        return Promise.reject();
+      }
```

Comment:
Just for fun, I asked Claude to review this and it reverted with this suggestion: 

```
if (!this.canSave) {
    // File cannot be saved.
    return Promise.reject(new Error('File cannot be saved')); // Add error message
}       

Feel free to ignore Claude! Else it all LGTM.

```